### PR TITLE
combine proxy vars from cluster-apps-operator

### PR DESCRIPTION
**Pull request by bavarianbidi**
## Checklist

- [x] Update changelog in CHANGELOG.md.

Towards https://github.com/giantswarm/roadmap/issues/1424

### CODE REVIEW

File: helm/chart-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -99,25 +100,27 @@ spec:
...
+        {{- if or (and $proxy.http $proxy.https $proxy.noProxy ) .Values.bootstrapMode.enabled }}
         env:
         {{- if .Values.bootstrapMode.enabled }}
         - name: KUBERNETES_SERVICE_HOST
           value: 127.0.0.1
         - name: KUBERNETES_SERVICE_PORT
           value: {{ .Values.bootstrapMode.apiServerPodPort | quote }}
         {{- end }}
-        {{- if .Values.proxy.http }}
+        {{- if and $proxy.noProxy $proxy.http $proxy.https }}
```
Comment (kubasobon) [inline helm/chart-operator/templates/deployment.yaml:111]: Will this change the behavior then? To where all of the `$proxy.*` variables have to be set to enable proxy at all?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,12 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+## Added
+
+- Support for running behind a proxy.
+  - `HTTP_PROXY`,`HTTPS_PROXY` and `NO_PROXY` are set as environment variables in `deployment/chart-operator` if defined in `values.yaml`.
```
Comment (kubasobon) [inline CHANGELOG.md:13]:
```suggestion
- `HTTP_PROXY`,`HTTPS_PROXY` and `NO_PROXY` are set as environment variables in `deployment/chart-operator` if defined in `values.yaml`.
```

**Review by kubasobon — APPROVED** [review]

File: helm/chart-operator/templates/deployment.yaml (hunk 1 above)
Comment (uvegla) [reply helm/chart-operator/templates/deployment.yaml:111]: I think `cluster-apps-operator` here: https://github.com/giantswarm/cluster-apps-operator/pull/290

Comment (bavarianbidi) [reply helm/chart-operator/templates/deployment.yaml:111]: The behavior itself wont change (beside the fact, that the proxy-vars are set in the `pod.spec` if the vars are set - and not if `proxy.enabled` equals `true`).

It's still possible to define proxy vars on the chart/app. But it's also possible to use the generated proxy variables from `cluster-apps-operator`.

If both proxy values (`cluster.proxy` and `proxy`) are set, the `proxy` values take precedence.

File: CHANGELOG.md (hunk 2 above)
Comment (bavarianbidi) [reply CHANGELOG.md:13]: i've set this intentionally but i will adjust the changelog to add some more details about precedence of `proxy` vs. `cluster.proxy`

File: helm/chart-operator/templates/deployment.yaml (hunk 1 above)
Comment (uvegla) [reply helm/chart-operator/templates/deployment.yaml:111]: What is the case for non CAPx clusters? This is a breaking change for them, right? I mean the values has to be aligned / where do these settings come from on legacy clusters?

Comment (bavarianbidi) [reply helm/chart-operator/templates/deployment.yaml:111]: @uvegla i wasn't aware that we have legacy clusters behind a corporate proxy. I've thought this is a new feature which is only implemented in new CAPI clusters.

The implementation will behave as follows (due some local testing):

* if `cluster.proxy` is defined, these values will be used
* if `proxy` is set and `cluster.proxy` is set, the configured `proxy` values take precedence
* if `proxy` is set and no `cluster.proxy` is set, the `proxy` values will be used.

So i think this isn't a breaking change on vintage clusters (but maybe i'm wrong?)

Comment (bavarianbidi) [reply helm/chart-operator/templates/deployment.yaml:111]: > Will this change the behavior then? To where all of the `$proxy.*` variables have to be set to enable proxy at all?

@kubasobon yes, you're right. Do you think this is a problem at all?

Comment (kubasobon) [reply helm/chart-operator/templates/deployment.yaml:111]: > > Will this change the behavior then? To where all of the `$proxy.*` variables have to be set to enable proxy at all?
> 
> @kubasobon yes, you're right. Do you think this is a problem at all?

I was not sure, but based on the answers here, I think we're safe :)

**Review by erkanerol — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (jkremser) [inline CHANGELOG.md:13]:
```suggestion
  - `HTTP_PROXY`,`HTTPS_PROXY`, `NO_PROXY` and their lower-case forms are set as environment variables in `deployment/chart-operator` if defined in `values.yaml`.
```

**Pull request merged**